proto/dolt/services/replicationapi/v1alpha1/replication.proto

```diff
@@ -0,0 +1,38 @@
+// Copyright 2023 Dolthub, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+package dolt.services.replicationapi.v1alpha1;
+
+option go_package = "github.com/dolthub/dolt/go/gen/proto/dolt/services/replicationapi/v1alpha1;replicationapi";
+
+service ReplicationService {
+  // Users and grants in Dolt are stored in in a
+  // `github.com/dolthub/go-mysql-server/sql/mysql_db.*MySQLDb` instance. This
+  // method is called by a primary on a standby instance in order to set its
+  // in-effect users and grants. Its primary payload is the serialized contents
+  // of the `*MySQLDb` instance at the primary, such that it can be applied
+  // with OverwriteUsersAndGrantData.
+  rpc UpdateUsersAndGrants(UpdateUsersAndGrantsRequest) returns (UpdateUsersAndGrantsResponse);
+}
+
+message UpdateUsersAndGrantsRequest {
+  // The contents of the *MySQLDb instance, as seen by a Persister
+  // implementation.
+  bytes serialized_contents = 1;
```

Comment:
Good point.

It's definitely implicit in the serialized contents, since those need to be forward compatible at the least, but when we get to replication scenarios, we get into more-likely-to-occur backward compatibility concerns. In particular, when a primary is newer than a replica, and we apply an upgraded version at the replica, the replica will interpret the data ignoring any fields it does not understand. The primary will have no way of knowing it is not actually up to date, and the replica, if it becomes a primary in the future, will replicate the users and grants without those previously applied newer features to all the replicas in the cluster.

A first pass proposal at solving something like that is to make the replica report its Dolt version, and to replicate the current users and grants to it if its dolt version changes. 

Thoughts?